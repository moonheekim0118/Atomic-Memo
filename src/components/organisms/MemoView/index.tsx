import React from 'react';
import TitleView from '../../molecules/TitleView';
import MainView from '../../molecules/MainView';
import memoData from '../../../model/memoData';
import Button from '../../atoms/Button';
import styled from 'styled-components';

interface Props {
  data: memoData;
  type: 'memos' | 'trash';
  onClickUpdate?: (e: React.MouseEvent) => void;
  onClickRemove?: (e: React.MouseEvent) => void;
}

const MemoView = ({ data, type, onClickUpdate, onClickRemove }: Props) => {
  return (
    <StyledMemoView>
      <TitleView text={data.title} time={data.time} />
      <MainView text={data.main} />
      <ButtonContainer>
        {type === 'memos' ? (
          <>
            <Button color="green" onClick={onClickUpdate}>
              Update
            </Button>
            <Button color="red" onClick={onClickRemove}>
              Remove
            </Button>
          </>
        ) : (
          <Button color="red" onClick={onClickRemove}>
            Remove from Trash
          </Button>
        )}
      </ButtonContainer>
    </StyledMemoView>
  );
};

const StyledMemoView = styled.section`
  height: 100%;
`;

const ButtonContainer = styled.div`
  position: absolute;
  bottom: 30px;
  right: 10%;
  width: 200px;

  display: flex;
  justify-content: space-between;
`;

export default MemoView;
