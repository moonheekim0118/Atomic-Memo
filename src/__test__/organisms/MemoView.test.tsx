import 'jsdom-global/register';
import React from 'react';
import MemoView from '../../components/organisms/MemoView';
import { mount } from 'enzyme';

describe('<MemoView/>', () => {
  let container = null;
  let mockData = {
    id: 'test',
    title: 'test',
    main: 'test',
    kind: 'test',
    time: 'test',
  };
  it('renders correctly', () => {
    container = mount(<MemoView data={mockData} type="memos" />);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
