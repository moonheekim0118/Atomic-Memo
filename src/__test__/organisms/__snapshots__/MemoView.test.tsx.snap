// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoView/> matches snpashot 1`] = `"<section class=\\"MemoView__StyledMemoView-sc-2dzfig-0 cOrwwB\\"><div class=\\"TitleView__StyledTitleView-sc-18c7sy9-0 bPlOFX\\"><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG title\\">test</span><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG small\\">test</span></div><div class=\\"MainView__StyledMainView-sc-1f2rj8k-0 iyqmxW\\"><span width=\\"auto\\" display=\\"block\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 ftDQbL normal\\">test</span></div><div class=\\"MemoView__ButtonContainer-sc-2dzfig-1 isNdqB\\"><button color=\\"green\\" class=\\"Button__StyledButton-hxb5gx-0 kaJctj\\">Update</button><button color=\\"red\\" class=\\"Button__StyledButton-hxb5gx-0 jwwHFX\\">Remove</button></div></section>"`;
